src/main/java/com/minecolonies/coremod/commands/colonycommands/DisableBarbarianSpawnsCommand.java

```diff
@@ -43,35 +53,63 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     }
 
     @Override
-    public void execute(
-                         @NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
     {
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
+        {
+            colony = (Colony) colonyObject;
+        }
 
-        if (sender instanceof EntityPlayer && !isPlayerOpped(sender))
+        boolean canHaveBarbEvents = false;
+        final Object disableSpawnsObject = argumentValueByActionArgumentNameMap.get("disableSpawns");
+        if (null != disableSpawnsObject)
         {
-            sender.sendMessage(new TextComponentString("Must be OP to use this command"));
+            final Boolean disableSpawnsBoolean = (Boolean) disableSpawnsObject;
+            canHaveBarbEvents = !disableSpawnsBoolean.booleanValue();
         }
 
-        final int colonyId;
+        executeShared(server, sender, colony, canHaveBarbEvents);
+    }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
         if (args.length == 0)
         {
             sender.sendMessage(new TextComponentString(NO_ARGUMENTS));
         }
 
+        final int colonyId;
+
         colonyId = getIthArgument(args, 0, -1);
 
         final Colony colony = ColonyManager.getColony(colonyId);
 
         if (colony == null)
         {
-            sender.sendMessage(new TextComponentString(NO_COLONY_FOUND_MESSAGE_ID));
+            final String noColonyFoundMessage = String.format(NO_COLONY_WITH_ID_FOUND_MESSAGE, colonyId);
+            sender.sendMessage(new TextComponentString(noColonyFoundMessage));
             return;
         }
 
-        final boolean setBarbsTo = Boolean.parseBoolean(args[1]);
+        final boolean canHaveBarbEvents = Boolean.parseBoolean(args[1]);
+
+        executeShared(server, sender, colony, canHaveBarbEvents);
+    }
+
+    private void executeShared(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final Colony colony, final boolean canHaveBarbEvents)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "server". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)